src/lib.rs

```diff
@@ -259,6 +259,25 @@ impl Discord {
 		check_status(result)
 	}
 
+    /// Create a role.
+    pub fn create_role(&self, server: ServerId) -> Result<Role>{
+        let response = request!(self, post, "/guilds/{}/roles", server);
+        Role::decode(try!(serde_json::from_reader(response)))
+    }
```

Comment:
Discord has released an [API update](https://github.com/hammerandchisel/discord-api-docs/commit/ee358f6f0870d3153f153222a2b62b1027f69536) to allow setting properties on create. The `EditRole` builder can be used here too now.